### ISSUE TITLE
lr_decay_epochs会在设置的warmup_steps之后才衰减学习率

### ISSUE THREAD

**Issue by judy19981229** [PaddlePaddle/PaddleX#1625]
## Checklist:

1. 查找[历史相关issue](https://github.com/PaddlePaddle/PaddleX/issues)寻求解答
2. 翻阅[FAQ常见问题汇总和答疑](https://github.com/PaddlePaddle/PaddleX/blob/develop/docs/FAQ/FAQ.md)
3. 确认bug是否在新版本里还未修复
4. 翻阅[PaddleX API文档说明](https://github.com/PaddlePaddle/PaddleX/tree/develop#paddlex-%E4%BD%BF%E7%94%A8%E6%96%87%E6%A1%A3)

## 描述问题
model.train(
    num_epochs=300,
    train_dataset=train_dataset,
    train_batch_size=8,
    eval_dataset=eval_dataset,
    learning_rate=0.000125,
    warmup_steps=1000,  
    warmup_start_lr=0.0,  
    lr_decay_epochs=[150, 240],
    lr_decay_gamma=0.1,
    save_interval_epochs=10,
    save_dir='output/yolov3_darknet53',
    pretrain_weights='IMAGENET',
    use_vdl=True)
设置lr_decay_epochs=[150, 240]，但是日志显示，学习率在196轮之后才衰减，而我的学习率正好在46轮达到了0.000125
2022-11-06 15:32:51 [INFO]	[TRAIN] Epoch=45/300, Step=2/22, loss_xy=2.984623, loss_wh=1.116557, loss_obj=9.226853, loss_cls=0.109391, loss=13.437424, lr=0.000121, time_each_step=1.0s, eta=1:36:41
2022-11-06 15:32:59 [INFO]	[TRAIN] Epoch=45/300, Step=12/22, loss_xy=2.694500, loss_wh=0.869965, loss_obj=7.968734, loss_cls=0.134891, loss=11.668091, lr=0.000122, time_each_step=0.84s, eta=1:21:16
2022-11-06 15:33:07 [INFO]	[TRAIN] Epoch=45/300, Step=22/22, loss_xy=2.746938, loss_wh=0.992552, loss_obj=8.021211, loss_cls=0.080223, loss=11.840923, lr=0.000124, time_each_step=0.77s, eta=1:14:29
2022-11-06 15:33:07 [INFO]	[TRAIN] Epoch 45 finished, loss_xy=2.6382122, loss_wh=1.047729, loss_obj=7.5985565, loss_cls=0.115582824, loss=11.400079 .
2022-11-06 15:33:16 [INFO]	[TRAIN] Epoch=46/300, Step=10/22, loss_xy=2.780983, loss_wh=0.698088, loss_obj=6.598033, loss_cls=0.069618, loss=10.146722, lr=0.000125, time_each_step=0.89s, eta=1:25:43
2022-11-06 15:33:23 [INFO]	[TRAIN] Epoch=46/300, Step=20/22, loss_xy=2.769047, loss_wh=1.177618, loss_obj=7.961426, loss_cls=0.111873, loss=12.019962, lr=0.000125, time_each_step=0.72s, eta=1:10:20
2022-11-06 15:33:24 [INFO]	[TRAIN] Epoch 46 finished, loss_xy=2.5912657, loss_wh=1.0137122, loss_obj=7.4151998, loss_cls=0.13078749, loss=11.150964 .
2022-11-06 16:20:40 [INFO]	[TRAIN] Epoch=194/300, Step=4/22, loss_xy=2.579414, loss_wh=0.566470, loss_obj=4.086326, loss_cls=0.012884, loss=7.245094, lr=0.000125, time_each_step=0.79s, eta=0:31:28
2022-11-06 16:20:47 [INFO]	[TRAIN] Epoch=194/300, Step=14/22, loss_xy=2.579310, loss_wh=0.592862, loss_obj=4.680823, loss_cls=0.020797, loss=7.873792, lr=0.000125, time_each_step=0.7s, eta=0:27:34
2022-11-06 16:20:53 [INFO]	[TRAIN] Epoch 194 finished, loss_xy=2.3993433, loss_wh=0.5818012, loss_obj=4.3942814, loss_cls=0.016528733, loss=7.391954 .
2022-11-06 16:20:56 [INFO]	[TRAIN] Epoch=195/300, Step=2/22, loss_xy=2.632890, loss_wh=0.885137, loss_obj=4.626435, loss_cls=0.022302, loss=8.166764, lr=0.000125, time_each_step=0.92s, eta=0:36:0
2022-11-06 16:21:04 [INFO]	[TRAIN] Epoch=195/300, Step=12/22, loss_xy=2.354830, loss_wh=0.626223, loss_obj=4.923774, loss_cls=0.013135, loss=7.917962, lr=0.000125, time_each_step=0.82s, eta=0:31:59
2022-11-06 16:21:10 [INFO]	[TRAIN] Epoch=195/300, Step=22/22, loss_xy=2.409441, loss_wh=0.588706, loss_obj=3.506605, loss_cls=0.005347, loss=6.510099, lr=0.000125, time_each_step=0.61s, eta=0:23:50
2022-11-06 16:21:11 [INFO]	[TRAIN] Epoch 195 finished, loss_xy=2.4828398, loss_wh=0.6444353, loss_obj=4.7860274, loss_cls=0.021661531, loss=7.934964 .
2022-11-06 16:21:19 [INFO]	[TRAIN] Epoch=196/300, Step=10/22, loss_xy=2.524044, loss_wh=0.522650, loss_obj=3.321416, loss_cls=0.018552, loss=6.386661, lr=0.000125, time_each_step=0.83s, eta=0:32:18
2022-11-06 16:21:26 [INFO]	[TRAIN] Epoch=196/300, Step=20/22, loss_xy=2.172996, loss_wh=0.821447, loss_obj=3.453926, loss_cls=0.012687, loss=6.461056, lr=0.000013, time_each_step=0.76s, eta=0:29:23
2022-11-06 16:21:28 [INFO]	[TRAIN] Epoch 196 finished, loss_xy=2.4182146, loss_wh=0.6325901, loss_obj=4.476193, loss_cls=0.014582312, loss=7.54158 .
2022-11-06 16:49:41 [INFO]	[TRAIN] Epoch=285/300, Step=2/22, loss_xy=2.569678, loss_wh=0.450463, loss_obj=4.841991, loss_cls=0.014350, loss=7.876482, lr=0.000013, time_each_step=0.8s, eta=0:4:45
2022-11-06 16:49:48 [INFO]	[TRAIN] Epoch=285/300, Step=12/22, loss_xy=2.349061, loss_wh=0.516972, loss_obj=4.492975, loss_cls=0.018212, loss=7.377220, lr=0.000013, time_each_step=0.69s, eta=0:3:58
2022-11-06 16:49:55 [INFO]	[TRAIN] Epoch=285/300, Step=22/22, loss_xy=2.291277, loss_wh=0.512787, loss_obj=3.853121, loss_cls=0.011805, loss=6.668990, lr=0.000013, time_each_step=0.69s, eta=0:3:52
2022-11-06 16:49:55 [INFO]	[TRAIN] Epoch 285 finished, loss_xy=2.3889563, loss_wh=0.49501994, loss_obj=3.854349, loss_cls=0.01675188, loss=6.755077 .
2022-11-06 16:50:04 [INFO]	[TRAIN] Epoch=286/300, Step=10/22, loss_xy=2.679095, loss_wh=0.527791, loss_obj=7.156189, loss_cls=0.095606, loss=10.458681, lr=0.000013, time_each_step=0.88s, eta=0:4:44
2022-11-06 16:50:12 [INFO]	[TRAIN] Epoch=286/300, Step=20/22, loss_xy=2.293489, loss_wh=0.465159, loss_obj=4.644278, loss_cls=0.052231, loss=7.455158, lr=0.000001, time_each_step=0.78s, eta=0:4:5
2022-11-06 16:50:14 [INFO]	[TRAIN] Epoch 286 finished, loss_xy=2.4394848, loss_wh=0.5289033, loss_obj=4.4378357, loss_cls=0.018377567, loss=7.4246016 .
## 复现

1. 您是否已经正常运行我们提供的[教程](https://github.com/PaddlePaddle/PaddleX/tree/develop/tutorials)？

2. 您是否在教程的基础上修改代码内容？还请您提供运行的代码

3. 您使用的数据集是？

4. 请提供您出现的报错信息及相关log

## 环境

1. 请提供您使用的PaddlePaddle和PaddleX的版本号
PP=2.3.2
paddlex=2.1.0

3. 请提供您使用的操作系统信息，如Linux/Windows/MacOS
使用AISTUDIO

5. 请问您使用的Python版本是？
Python3

6. 请问您使用的CUDA/cuDNN的版本号是？

**Comment by lailuboy** [comment]
是的，会先warmup，你现在两个参数设置的epoch数是多少？

**Comment by judy19981229** [comment]
我的样本数比较少，有930张图片训练，我设置的warmup大概需要10个epoch
